sound/soc/sof/ipc4-topology.c

```diff
@@ -1313,6 +1313,10 @@ sof_ipc4_update_resource_usage(struct snd_sof_dev *sdev, struct snd_sof_widget *
 	pipeline = pipe_widget->private;
 	pipeline->mem_usage += total;
 
+	/* set pipeline direction */
+	pipeline->msg.extension |= SOF_IPC4_GLB_PIPE_EXT_DIRECTION_SET(0x1);
+	pipeline->msg.extension |= SOF_IPC4_GLB_PIPE_EXT_DIRECTION(dir);
```

Comment:
Document what dir value is what direction, I assume playback is 0, capture is 1?
and to be safe with the integer:
```c
	pipeline->msg.extension |= SOF_IPC4_GLB_PIPE_EXT_DIRECTION(!!dir);
```
Another way to put this could be to
#define SOF_IPC4_GLB_PIPE_EXT_DIR_MASK	GENMASK(25, 24)
#define SOF_IPC4_GLB_PIPE_EXT_DIR_SET(dir) ((dir + 1) << 24)

so:
0 means it is not provided
1 means it is playback direction
2 means capture
3 for later use